src/main/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoProcessor.java

```diff
@@ -174,6 +173,10 @@ public void onResponse(final Datasource datasource) {
 
             @Override
             public void onFailure(final Exception e) {
+                if (e instanceof IndexNotFoundException) {
```

Comment:
is check on exception class enough here, should we also check on a message to minimize chance of false positive?

Reply:
Check on exception class is enough. No other cases. It is only for error message display so not a big issue even if there is false positive.